scipy/interpolate/_interpolate.py

```diff
@@ -2452,10 +2463,18 @@ class RegularGridInterpolator:
     # this class is based on code originally programmed by Johannes Buchner,
     # see https://github.com/JohannesBuchner/regulargrid
 
+    SPLINE_DEGREE_MAP = {"slinear": 1, "cubic": 3, "quintic": 5, }
+    SPLINE_METHODS = list(SPLINE_DEGREE_MAP.keys())
+    ALL_METHODS = ["linear", "nearest"] + SPLINE_METHODS
+
     def __init__(self, points, values, method="linear", bounds_error=True,
                  fill_value=np.nan):
-        if method not in ["linear", "nearest"]:
+        if method not in self.ALL_METHODS:
             raise ValueError("Method '%s' is not defined" % method)
+        elif method in self.SPLINE_METHODS:
+            self._validate_not_complex(points, method)
+            self._validate_not_complex(values, method)
```

Comment:
Is this needed? 
ISTM this is a historic artifact from initial iterations of gh-7903 which were using UnivariateSpline. Now that it uses `make_interp_spline / BSpline`, complex-values `values` should just work, I'd think. See also gh-2650. 
This can be done is a follow-up PR however.

Reply:
> Now that it uses make_interp_spline / BSpline, complex-values values should just work,
I didn't know that. I will try to remove these checks and add tests for complex input.

Reply:
I removed the validator and added new methods to `test_complex`. I confirmed the test passed.